coalib/files/Fileproxy.py

```diff
@@ -0,0 +1,89 @@
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq("filename")
+class Fileproxy:
+    """
+    The file proxy is a wrapper object to wrap every file. The file proxy is an
+    object that contains properties about the file currently processed (such as
+    the filename). It contains:
+
+        * The filename, this is the starting point for every other attribute.
+        * The content of the file as a string.
+        * The content of the file, line-splitted.
+
+    The equality of the object is checked just based on its filename.
+
+    To initialize a Fileproxy object for a file:
+
+    >>> fp = Fileproxy("./tests/files/FileproxyTestFiles/testfile.txt")
+
+    Now we can get the filename:
+
+    >>> fp.filename
+    './tests/files/FileproxyTestFiles/testfile.txt'
+
+    The number of lines in the file:
+
+    >>> len(fp)
+    1
+
+    Iterate through the lines of the file:
+
+    >>> for line in fp:
+    ...     print(line)
+    Old line
+    <BLANKLINE>
+
+    Get contents of the file as a string:
+
+    >>> fp.content
+    'Old line\\n'
+
+    Get all lines of the file as a list:
+    >>> fp.lines
+    ['Old line\\n']
+    """
+
+    def __init__(self, filename):
```

Comment:
Few general comments here:
 - It makes sense to do the equality check with os.path.abspath(filename). Because ./test.py and ./abc/../test.py are actually the same ?
 - Also have a `absolute_filename` like thing for the absolute path to the file ?
 - Please mention that this does not be "closed" like a normal file pointer